src/Pages/CropCycleTimeline.js

```diff
@@ -0,0 +1,317 @@
+import {
+  FormControl,
+  FormGroup,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  Snackbar,
+  Typography,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { Alert } from '@material-ui/lab';
+import AmsTimeline from 'components/common/AmsTimeline';
+import { getCropCyclesByField } from 'dataclients/CropCyclesClient';
+import { getFieldsByLocation } from 'dataclients/FieldsClient';
+import { getAllInspectionsByCropCycle } from 'dataclients/InspectionsClient';
+import { getLocations } from 'dataclients/LocationsClient';
+import React, { useEffect, useState } from 'react';
+
+const useStyles = makeStyles((theme) => ({
+  gridItem: {
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+  formControl: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    minWidth: 300,
+    maxWidth: 300,
+  },
+  submitButton: {
+    margin: theme.spacing(3),
+    width: 'fit-content',
+    minWidth: '150px',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+  menuItem: {
+    maxWidth: 300,
+    whiteSpace: 'normal',
+  },
+  helpText: {
+    fontSize: theme.typography.pxToRem(14),
+    color: theme.palette.text.secondary,
+  },
+}));
+
+/**
+ * Renders timeline for a crop cycle, showing the processes and the
+ * inspections associated with it.
+ */
+const CropCycleTimeline = () => {
+  const classes = useStyles();
+  const [locationCode, setLocationCode] = useState('');
+  const [locations, setLocations] = useState([]);
+  const [fieldId, setFieldId] = useState('');
+  const [fields, setFields] = useState([]);
+  const [cropCycleId, setCropCycleId] = useState('');
+  const [cropCycles, setCropCycles] = useState([]);
+  const [inspections, setInspections] = useState([]);
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [alertSeverity, setAlertSeverity] = useState('');
+
+  const handleAlertClose = (_event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setAlertStatus(false);
+  };
+
+  const showAlert = (message, severity) => {
```

Comment:
function documentation?

Reply:
Common method used across all the pages, trivial.